src/remote.cc

```diff
@@ -537,18 +537,31 @@ String get_user_name()
     return getenv("USER");
 }
 
+String session_path(StringView session)
+{
+    auto slash_count = std::count(session.begin(), session.end(), '/');
+    if (slash_count > 0)
+        throw runtime_error{"session names cannot have slashes"};
+    StringView xdg_runtime_dir = getenv("XDG_RUNTIME_DIR");
+    if (xdg_runtime_dir.empty())
+    {
+        // set sticky bit on the shared kakoune directory
+        make_directory(format("{}/kakoune", tmpdir()), 01777);
+        make_directory(format("{}/kakoune/{}", tmpdir(), get_user_name()), 0711);
```

Comment:
I find it surprising that the `session_path` function would create directories as a side effect, why is the sticky bit needed here ?

Reply:
The sticky bit was already there previously.  I looked it up, and it prevents users from renaming and deleting each others' sockets in `/tmp/kakoune`, which normally only requires directory write permission.